### ISSUE TITLE
Spin up new security service API

### ISSUE THREAD

**Issue by plittlewood-rpt** [reapit/foundations#8974]
**Background context**
#9167 should be completed before this ticket

To allow support for a more enhanced security offering that can back other group products, we should spin up a brand new API to serve this purpose. Please refer to internal implementation document for project summary

This item relates to the scaffolding of the new API only

**Specification**
- Spin up new Reapit.Services.Security API – this should be a containerised .NET 6 service and use the relevant project template, rather than being serverless as it will be hit very frequently. For now this will interact with the PermissionSet and PermissionGrp entities in AgencyCloud, but will eventually be superseded by a DynamoDb (or similar) NoSQL backend
- DevOps project has already been created and should be ready to go
- Add placeholder endpoints as follows to SecurityController, with their respective Command/Query/Request counterparts – there should be no functionality attached to their handlers for now, they will be fleshed out in subsequent tickets:

General
  -	GET / -> GetResolvedUserPermissionsQuery (will take an access token)

Role Management (AC Permission Groups under the hood)
  -	GET /security/roles -> GetRolesRequest / GetRolesQuery / RoleModel[]
  -	GET /security/roles/{id} -> GetRoleByIdRequest / GetRoleByIdQuery / RoleModel
  -	POST /security/roles -> CreateRoleCommand / CreateRoleModel
  -	PUT /security/roles/{id} -> UpdateRoleCommand / UpdateRoleModel
  -	DELETE /security/roles/{id} -> DeleteRoleCommand
  -	GET /security/roles/{id}/members -> GetRoleMembersRequest / GetRoleMembersQuery / RoleMemberModel
  -	POST /security/roles/{id}/members -> CreateRoleMemberCommand (to attach users/apps) / CreateRoleMemberModel
  -	DELETE /security/roles/{id}/members/{memberId} -> DeleteRoleMemberCommand (to remove role assignment from users/apps)
  -	GET /security/roles/{id}/policies -> GetRolePoliciesRequest / GetRolePoliciesQuery / PolicyModel[]

Policy Management (AC Permission Sets under the hood)
  -	GET /security/policies -> GetPoliciesRequest / GetPoliciesQuery / PolicyModel[]
  -	GET /security/policies/{id} -> GetPolicyByIdRequest / GetPolicyByIdQuery / PolicyModel
  -	GET /security/policies/{id}/members -> GetPolicyMembersRequest / GetPolicyMembersQuery / PolicyMemberModel
  -	POST /security/policies/{id}/members -> CreatePolicyMemberCommand (to attach roles only) / CreatePolicyMemberModel
  -	DELETE /security/policies/{id}/members -> DeletePolicyMemberCommand
  -	POST /security/policies/ -> CreatePolicyCommand / CreatePolicyModel
  -	PUT /security/policies/{id} -> UpdatePolicyCommand / UpdatePolicyModel
  -	DELETE /security/policies/{id} -> DeletePolicyCommand

- Use a mapping layer as per other services to map between requests/queries. Build the layer so it’s ready for the full models later
- New scope security.write should be used to protect write endpoints. Read endpoints will not be protected by a scope as it's effectively configuration data (we'll have identity checks in the handler)
